### PR TITLE
[#5304] doc(core): add event listener document

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?

add event listener document

### Why are the changes needed?

Fix: #5304 

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?
just document

### CODE REVIEW

**Comment by FANNG1** [comment]
@jerryshao @yuqi1129 @jerqi please help to review when you have time, thanks

File: docs/gravitino-server-config.md [hunk 1]
```diff
@@ -113,16 +113,23 @@ To leverage the event listener, you must implement the `EventListenerPlugin` int
...
+| metalake operation                  | `CreateMetalakeEvent`, `AlterMetalakeEvent`, `DropMetalakeEvent`, `LoadMetalakeEvent`, `ListMetalakeEvent`, `CreateMetalakeFailureEvent`, `AlterMetalakeFailureEvent`, `DropMetalakeFailureEvent`, `LoadMetalakeFailureEvent`, `ListMetalakeFailureEvent`                                                                                                                                                                     | 0.5.0            |
+| Iceberg REST server table operation | `IcebergCreateTableEvent`, `IcebergUpdateTableEvent`, `IcebergDropTableEvent`, `IcebergLoadTableEvent`, `IcebergListTableEvent`, `IcebergTableExistsEvent`, `IcebergRenameTableEvent`, `IcebergCreateTableFailureEvent`, `IcebergUpdateTableFailureEvent`, `IcebergDropTableFailureEvent`, `IcebergLoadTableFailureEvent`, `IcebergListTableFailureEvent`, `IcebergRenameTableFailureEvent`, `IcebergTableExistsFailureEvent` | 0.7.0-incubating |
+
+Pre-event is generated before Iceberg REST server table operations only.
```
Comment (jerryshao) [inline docs/gravitino-server-config.md:128]: This sentence is hard to understand, what do you want to express?

Comment (FANNG1) [reply docs/gravitino-server-config.md:128]: Pre-event is only generated before Iceberg REST server table operations, the other operation doesn't generate pre-event

Comment (jerryshao) [reply docs/gravitino-server-config.md:128]: You can explain well in here, but why can't you improve your phrase in the doc?

File: docs/gravitino-server-config.md [hunk 2]
```diff
@@ -113,16 +113,23 @@ To leverage the event listener, you must implement the `EventListenerPlugin` int
 
 #### Event
 
-Gravitino triggers a specific event upon the completion of the operation, with varying events being generated for different operations.
+Gravitino triggers a pre-event before the operation, a post-event after the completion of the operation and a failure event after the operation failed.
```
Comment (jerryshao) [inline docs/gravitino-server-config.md:116]: You can have two tables, one for post-event, one for pre-event, and clearly add a heading to each table.

File: docs/gravitino-server-config.md (hunk 1 above)
Comment (FANNG1) [reply docs/gravitino-server-config.md]: updated

File: docs/gravitino-server-config.md [hunk 3]
```diff
@@ -136,7 +143,7 @@ The plugin provides several operational modes for how to process event, supporti
  
 - **ASYNC_ISOLATED**: Events are processed asynchronously, with each listener having its own dedicated queue and dispatcher thread. This approach offers better isolation but at the expense of multiple queues and dispatchers.
 
-For more details, please refer to the definition of the plugin.
+When processing pre-event, you could throw up `ForbiddenException` to skip the execution of subsequent operation. For more details, please refer to the definition of the plugin.
```
Comment (jerryshao) [inline docs/gravitino-server-config.md:146]: "throw a ... to skip the following executions."

Comment (FANNG1) [reply docs/gravitino-server-config.md]: updated

File: docs/gravitino-server-config.md [hunk 4]
```diff
@@ -113,16 +113,23 @@ To leverage the event listener, you must implement the `EventListenerPlugin` int
...
-| catalog operation  | `CreateCatalogEvent`, `AlterCatalogEvent`, `DropCatalogEvent`, `LoadCatalogEvent`, `ListCatalogEvent`, `CreateCatalogFailureEvent`, `AlterCatalogFailureEvent`, `DropCatalogFailureEvent`, `LoadCatalogFailureEvent`, `ListCatalogFailureEvent`                                 |
-| metalake operation | `CreateMetalakeEvent`, `AlterMetalakeEvent`, `DropMetalakeEvent`, `LoadMetalakeEvent`, `ListMetalakeEvent`, `CreateMetalakeFailureEvent`, `AlterMetalakeFailureEvent`, `DropMetalakeFailureEvent`, `LoadMetalakeFailureEvent`, `ListMetalakeFailureEvent`                       |
+| Operation type                      | Post-event                                                                                                                                                                                                                                                                                                                                                                                                                    | Since Version    |
```
Comment (jerryshao) [inline docs/gravitino-server-config.md:118]: Add a new heading like `##### Pre-event` and `##### Post-event`.

Comment (FANNG1) [reply docs/gravitino-server-config.md:120]: updated

**Review by jerryshao — APPROVED** [review]

**Pull request merged**